### PR TITLE
[FAB-18125] Remove removeStorage flag from channel participation API

### PR DESCRIPTION
**Pull request by stephyee**
#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

Removes removeStorage flag from channel participation API

#### Related issues
[FAB-18125](https://jira.hyperledger.org/browse/FAB-18125)

### CODE REVIEW

File: orderer/common/channelparticipation/restapi.go [hunk 1]
```diff
@@ -279,12 +276,7 @@ func (h *HTTPHandler) serveRemove(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	removeStorage, err := h.extractRemoveStorageQuery(req, resp)
-	if err != nil {
-		return
-	}
-
-	err = h.registrar.RemoveChannel(channelID, removeStorage)
+	err = h.registrar.RemoveChannel(channelID)
```
Comment (wlahti) [inline orderer/common/channelparticipation/restapi.go:278]: The comment for this function still references removeStorage. (it won't let me add the comment to the actual code but it's a few lines above here)

**Review by wlahti — COMMENTED** [review]
Looks good other than one stale comment for serveRemove.

File: orderer/common/channelparticipation/restapi.go (hunk 1 above)
Comment (stephyee) [reply orderer/common/channelparticipation/restapi.go:278]: 🧹

**Review by wlahti — DISMISSED** [review]
Looks good to me. Thanks, TIffany! @jyellick Can you review this PR when you get a chance?

File: orderer/common/multichannel/registrar.go [hunk 2]
```diff
@@ -597,7 +597,7 @@ func (r *Registrar) createFollower(
 
 // RemoveChannel instructs the orderer to remove a channel.
 // Depending on the removeStorage parameter, the storage resources are either removed or archived.
-func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
+func (r *Registrar) RemoveChannel(channelID string) error {
```
Comment (tock-ibm) [inline orderer/common/multichannel/registrar.go]: Update the comment

**Comment by tock-ibm** [comment]
I had one minor comment, otherwise this looks good to me.

File: orderer/common/multichannel/registrar.go (hunk 2 above)
Comment (stephyee) [reply orderer/common/multichannel/registrar.go]: 🧹

**Review by jyellick — APPROVED** [review]

**Pull request merged**